src/dbnode/integration/large_tiles_test.go

```diff
@@ -0,0 +1,163 @@
+		nsOpts         = namespace.NewOptions().
+				SetRetentionOptions(rOpts).
+				SetIndexOptions(idxOpts).
+				SetColdWritesEnabled(true)
+	)
+
+	srcNs, err := namespace.NewMetadata(testNamespaces[0], nsOpts)
+	require.NoError(t, err)
+	trgNs, err := namespace.NewMetadata(testNamespaces[1], nsOpts)
+	require.NoError(t, err)
+
+	testOpts := NewTestOptions(t).
+		SetNamespaces([]namespace.Metadata{srcNs, trgNs}).
+		SetWriteNewSeriesAsync(true)
+
+	testSetup := newTestSetupWithCommitLogAndFilesystemBootstrapper(t, testOpts)
+	defer testSetup.Close()
+
+	reporter := xmetrics.NewTestStatsReporter(xmetrics.NewTestStatsReporterOptions())
+	scope, closer := tally.NewRootScope(
+		tally.ScopeOptions{Reporter: reporter}, time.Millisecond)
+	defer closer.Close()
+	testSetup.SetStorageOpts(testSetup.StorageOpts().SetInstrumentOptions(
+		instrument.NewOptions().SetMetricsScope(scope)))
+
+	storageOpts := testSetup.StorageOpts()
+	testSetup.SetStorageOpts(storageOpts)
+
+	// Start the server.
+	log := storageOpts.InstrumentOptions().Logger()
+	require.NoError(t, testSetup.StartServer())
+
+	// Stop the server.
+	defer func() {
+		require.NoError(t, testSetup.StopServer())
+		log.Debug("server is now down")
+	}()
+
+	start := time.Now()
+	session, err := testSetup.M3DBClient().DefaultSession()
+	require.NoError(t, err)
+	nowFn := testSetup.NowFn()
+
+	tags := []ident.Tag{
+		ident.StringTag("__name__", "cpu"),
+		ident.StringTag("job", "job1"),
+	}
+
+	dpTimeStart := nowFn().Truncate(indexBlockSize).Add(-3 * indexBlockSize)
+	dpTime := dpTimeStart
+
+	// Write test data.
+	for a := 0.0; a < 20.0; a++ {
+		err = session.WriteTagged(srcNs.ID(), ident.StringID("foo"), ident.NewTagsIterator(ident.NewTags(tags...)), dpTime, 42.1+a, xtime.Second, nil)
+		require.NoError(t, err)
+		dpTime = dpTime.Add(10 * time.Minute)
+	}
+	log.Info("test data written", zap.Duration("took", time.Since(start)))
+
+	log.Info("waiting till data is cold flushed")
+	start = time.Now()
+	expectedNumWrites := int64(20)
+	flushed := xclock.WaitUntil(func() bool {
+		counters := reporter.Counters()
+		counter, ok := counters["database.series.cold-writes"]     // Wait until data is written
+		warmData, ok := counters["database.flushWarmData.success"] // Wait until data is flushed
+		return ok && counter == expectedNumWrites && warmData >= expectedNumWrites
+	}, time.Minute)
+	require.True(t, flushed)
+	log.Info("verified data has been cold flushed", zap.Duration("took", time.Since(start)))
+
+	aggOpts := storage.AggregateTilesOptions{
+		Start: dpTimeStart,
+		End:   time.Now(),
+		Step:  time.Hour,
+	}
+
+	// Retry aggregation as persist manager could be still locked by cold writes.
+	// TODO: Remove retry when a separate persist manager will be implemented.
+	for retries := 0; retries < 10; retries++ {
+		err = testSetup.DB().AggregateTiles(storageOpts.ContextPool().Get(), srcNs.ID(), trgNs.ID(), aggOpts)
```

Comment:
Would it be possible to call this via RPC instead, for a more comprehensive coverage?